src/core/emergency-action.c

```diff
@@ -32,6 +44,15 @@ void emergency_action(
         assert(action >= 0);
         assert(action < _EMERGENCY_ACTION_MAX);
 
+        /* Is the special shutdown target active or queued? If so, we are in shutdown state */
+        if ((manager_state(m) == MANAGER_STOPPING) &&
+            ((action == EMERGENCY_ACTION_REBOOT) ||
+             (action == EMERGENCY_ACTION_POWEROFF))) {
+                log_info("EmergencyAction: Shutdown is already active Skipping %s request",
```

Comment:
hmm, so manager_state() does a lot of things internally, most of which aren't interesting here. I think I would make the test differently:

```c
Unit *u;
u = manager_get_unit(m, SPECIAL_SHUTDOWN_TARGET);
if (u && unit_active_or_pending(u) &&
        IN_SET(action, EMERGENCY_ACTION_REBOOT, EMERGENCY_ACTION_POWEROFF, EMERGENCY_ACTION_EXIT)) {
…
```

Also, I think this should be log_notice(), not log_info()

Reply:
ANd yeah, EMERGENCY_ACTION_EXIT should be covered too here, it's just another way to shutdown (one you would only use in a container, but that's a detail)